tests/conftest.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--runslow",
+        action="store",
+        default="no",
+        choices=("yes", "no", "only"),
+        help="whether to run slow tests",
+    )
+
+
+def pytest_configure(config):
+    config.addinivalue_line("markers", "slow: mark test as slow to run")
+
+
+def pytest_collection_modifyitems(config, items):
+    if config.getoption("--runslow") == "no":
+        skip_slow = pytest.mark.skip(reason="need --runslow option to run")
+        for item in items:
+            if "slow" in item.keywords:
+                item.add_marker(skip_slow)
+    elif config.getoption("--runslow") == "only":
```

Comment:
I cannot remember if there is any validation of these values. Would it make sense to raise an exception here if the user has set it to something other than yes/no/only?

Reply:
This is already validated by the `choices` argument to addoption. The options appear in `pytest --help` and you get the following error message if you mistype:

```
pytest: error: argument --runslow: invalid choice: 'maybe' (choose from 'yes', 'no', 'only')
```